src/lib/widgets/layers-toggle/layers-toggle.component.ts

```diff
@@ -1,16 +1,32 @@
-import { Component, OnInit, Input } from '@angular/core';
-
+import { Component, OnInit, Input/*, Pipe*/ } from '@angular/core';
 import { EsriMapService } from './../../core/index';
-
+/*@Pipe({
+  name: 'reverse'
+})
+export class ReversePipe {
+  transform(value:any) {
+    return value.slice().reverse();
+  }
+}*/
 @Component({
   selector: 'esri-layersToggle',
   template: `
     <div esri-customWidget [position]="position">
       <!-- Only display if there is a map and it has layers -->
       <div *ngIf="map?.layers.length > 0" class="esri-layers-toggle">
         <div *ngFor="let layer of map.layers.toArray()">
-          <input type="checkbox" [attr.checked]="layer.visible" (click)="onCheck($event, layer)" /> {{layer.title}}
+          <input type="checkbox" [checked]="layer.visible" (click)="onCheck($event, layer)" /> {{layer.title}}
           <a href="javascript:void(0)" (click)="onZoomLayer(layer)">Zoom</a>
+          <div *ngIf="layer.sublayers && layer.sublayers.items">
+            <div *ngFor="let lay of layer.sublayers.items.slice().reverse()">
+              &nbsp;&nbsp;&nbsp;<input type="checkbox" [checked]="lay.visible" (click)="onCheck($event, lay)" /> {{lay.title}}
+              <div *ngIf="lay.sublayers && lay.sublayers.items">
```

Comment:
Currently, the way that this is structured it would mean that the library supports up to three levels of layers (layer, sub-layer 1, sub-layer 2). If there was a 3rd sub-layer it would be ignored. It would be better if this were dynamic enough to handle as many sub-layer levels as is needed.

What you could do is create a separate component called Layers that is in charge of displaying an array of layers. The Layers component would take an Input of an array of layers. In the template for Layers if a given layer has sublayers then it would pass the array of sublayers into another instance of the Layers component. This would allow it to traverse all the way through the layers and sublayers to display as many levels as are provided.